core/src/main/java/feign/ReflectiveFeign.java

```diff
@@ -160,21 +167,16 @@ public String toString() {
     protected final MethodMetadata metadata;
     private final Map<Integer, Expander> indexToExpander = new LinkedHashMap<Integer, Expander>();
 
-    private BuildTemplateByResolvingArgs(MethodMetadata metadata) {
+    private BuildTemplateByResolvingArgs(MethodMetadata metadata, Factory paramExpanderFactory) {
       this.metadata = metadata;
+      checkNotNull(paramExpanderFactory, "paramExpanderFactory");
       if (metadata.indexToExpanderClass().isEmpty()) {
         return;
       }
       for (Entry<Integer, Class<? extends Expander>> indexToExpanderClass : metadata
```

Comment:
That's pretty simple and probably effective. For the sake of discussion, it would be possible to support `indexToExpander` and `indexToExpanderClass` simultaneously, preferring the former, but in practice deprecating `indexToExpanderClass` should mean that any `Contract` implementation using the new form would use it exclusively. The only exception to that might be if there is some composite `Contract` out there, but such an object could just as easily handle the merging of `expanderClass`es with `expander`s. Unless I hear differently I will implement per your thoughts above... however, until you tell me otherwise I am going to approach as `indexToExpander` being empty rather than `null` because that feels to me more  in keeping with the feel of the `MethodMetadata` API (no mutator method for `indexToExpanderClass`). Unless you want to do something wild like return `null` from `#indexToExpander()` if `#indexToExpanderClass()` is not empty, but that starts to suggest other things that become more and more complicated.